frontend/src/components/pages/FridgeManagement/FridgeCheckIns/EditCheckin.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  Button,
+  Center,
+  CloseButton,
+  Container,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  HStack,
+  Input,
+  Spinner,
+  Text,
+  Textarea,
+  useToast,
+} from "@chakra-ui/react";
+import { format, isAfter, isEqual, parse } from "date-fns";
+import React, { useEffect, useState } from "react";
+import { useHistory, useParams } from "react-router-dom";
+
+import CheckInAPIClient from "../../../../APIClients/CheckInAPIClient";
+import * as Routes from "../../../../constants/Routes";
+import { UpdateCheckInFields } from "../../../../types/CheckInTypes";
+import ErrorMessages from "./ErrorMessages";
+
+const EditCheckInPage = (): JSX.Element => {
+  const { id } = useParams<{ id: string }>();
+  const [date, setDate] = useState<string>("");
+  const [startTime, setStartTime] = useState<string>("");
+  const [endTime, setEndTime] = useState<string>("");
+  const [notes, setNotes] = useState<string | undefined>("");
+  const [timeRangeError, setTimeRangeError] = useState("");
+  const toast = useToast();
+  const history = useHistory();
+
+  const getCheckInData = async () => {
+    const checkInResponse = await CheckInAPIClient.getCheckInsById(id);
+    setDate(checkInResponse.startDate);
+    setStartTime(format(new Date(checkInResponse.startDate), "kk:mm"));
+    setEndTime(format(new Date(checkInResponse.endDate), "kk:mm"));
+    setNotes(checkInResponse.notes);
+  };
+
+  const validateForm = () => {
+    let newError = "";
+    let valid = true;
+    const parsedStartTime = parse(startTime, "kk:mm", new Date());
+    const parsedEndTime = parse(endTime, "kk:mm", new Date());
+
+    if (isAfter(parsedStartTime, parsedEndTime)) {
+      valid = false;
+      newError = ErrorMessages.endTimeBeforeStartTime;
+    }
+
+    setTimeRangeError(newError);
+    return valid;
+  };
+
+  const onSaveClick = async () => {
+    if (!validateForm()) {
+      return;
+    }
+
+    const checkInData: UpdateCheckInFields = {
+      startDate: parse(startTime, "kk:mm", new Date(date)).toString(),
+      endDate: parse(endTime, "kk:mm", new Date(date)).toString(),
+      notes,
+    };
+
+    const res = await CheckInAPIClient.updateCheckInById(id, checkInData);
+
+    if (!res.id) {
```

Comment:
a workaround i found is to check for `!res.id` instead of `res`. but we could change `updateCheckInById` to return `false` on error so that it is consistent with the rest of the code.